### DISCUSSION TITLE
http2dubbo upstream connect error or disconnect/reset before headers. reset reason: protocol error

### DISCUSSION THREAD

**Discussion by paxxie2** [higress-group/higress#1056] [Q&A]
1. 目标：测试服务器上， http 能调用 dubbo 服务， 让自定义插件能间接调用现有的 dubbo 业务服务，实现自定义网关功能。
2. 环境： 一个k8s集群， dubbo 业务服务以 pod 的形式部署到 K8s 集群， 使用 nacos 作为 dubbo 服务的注册中心；一个自建 higress 网关，放到测试环境， console 是 1.4.1 ， 其他组件是 1.4.0
3. 实验步骤：k8s集群中的 dubbo 服务在pod重启后，能本地开发电脑中用 dubbo 客户端访问测试环境的 k8s 集群中的 dubbo 服务；测试环境，配置 higress 的 http 转 dubbo， 使用 higress 网关作为流量入口，http 调用 dubbo 接口。
4. higress 配置：
4.1 控制台创建服务来源
4.2 控制台创建路由
4.3 打开 higress 使用的nacos， 里面配置CRD
5. 关键配置截图
5.1 服务来源
![image](https://github.com/alibaba/higress/assets/20533911/3cd7b89e-1291-41ce-ae25-710693d103b6)
5.2 路由配置
![image](https://github.com/alibaba/higress/assets/20533911/c0aafbdc-9afd-41ce-91c4-32206e1d115c)
5.3 http2dubbo 的CRD配置，图片无法配置，贴文本
命名空间： higress-system
nacos Data ID: http2rpcs.demo
nacos Group: DEFAULT_GROUP
yaml配置
`
apiVersion: networking.higress.io/v1
kind: Http2Rpc
metadata:
  name: httproute-http2rpc-demo
  namespace: higress-system
spec:
  dubbo: 
    service: com.隐藏.demo.dubbo.api.DemoService
    version: 1.0.0
    group: betaA
    methods: 
    - serviceMethod: sayHello
      headersAttach: "*"
      httpMethods: 
      - "GET"
      httpPath: "/dubbo/hello"
      params:
      - paramKey: p
        paramSource: QUERY
        paramType: "java.lang.String"
`
6. 测试接口
url:
`
http://arch-ad-management-test-1.隐藏.net/dubbo/hello/sayHello?name=xx
`
浏览器响应：
![image](https://github.com/alibaba/higress/assets/20533911/228360f2-041e-4967-9d08-65e2223ba7d9)
网关日志：
`
[Envoy (Epoch 0)] [2024-06-25 09:35:37.646][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] start http call: /api/test/plugin
[Envoy (Epoch 0)] [2024-06-25 09:35:37.657][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] http call status code: 200
[Envoy (Epoch 0)] [2024-06-25 09:35:37.658][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] http call response : {"sourceType":"gaia","code":"200","message":"OK","requestId":"f09cdd7b9d7043ca80e7f8d869120c17.206.17193080960108089","success":true,"data":"gaia-demo-plugin"}
[Envoy (Epoch 0)] [2024-06-25 09:35:37.658][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] get status: 200, request body: {"sourceType":"gaia","code":"200","message":"OK","requestId":"f09cdd7b9d7043ca80e7f8d869120c17.206.17193080960108089","success":true,"data":"gaia-demo-plugin"}
{"authority":"arch-ad-management-test-1.隐藏.net","bytes_received":"0","bytes_sent":"87","downstream_local_address":"172.20.0.7:80","downstream_remote_address":"172.16.110.32:62646","duration":"25","istio_policy_status":"-","method":"GET","path":"/dubbo/hello/sayHello?name=xx","protocol":"HTTP/1.1","request_id":"3ac4bcfb-30ed-48ec-bb24-292666a1ddec","requested_server_name":"-","response_code":"502","response_flags":"UPE","route_name":"gaia-demo-http2dubbo","start_time":"2024-06-25T09:35:37.645Z","trace_id":"a64e9e1c3ed881e0126144732374a85c","upstream_cluster":"outbound|20880||providers:com.隐藏.demo.dubbo.api.DemoService::betaA.DEFAULT-GROUP.3e3dcada-7c4a-4728-b4c4-1cfdb5b7d2d1.nacos","upstream_host":"10.244.36.242:20880","upstream_local_address":"172.20.0.7:35054","upstream_service_time":"-","upstream_transport_failure_reason":"-","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36","x_forwarded_for":"172.16.110.32"}
[Envoy (Epoch 0)] [2024-06-25 09:35:50.099][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] start http call: /api/test/plugin
[Envoy (Epoch 0)] [2024-06-25 09:35:50.107][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] http call status code: 200
[Envoy (Epoch 0)] [2024-06-25 09:35:50.107][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] http call response : {"sourceType":"gaia","code":"200","message":"OK","requestId":"f09cdd7b9d7043ca80e7f8d869120c17.207.17193081084606229","success":true,"data":"gaia-demo-plugin"}
[Envoy (Epoch 0)] [2024-06-25 09:35:50.107][51][error][wasm] wasm log higress-system.http-calss-1.0.0: [http-call] get status: 200, request body: {"sourceType":"gaia","code":"200","message":"OK","requestId":"f09cdd7b9d7043ca80e7f8d869120c17.207.17193081084606229","success":true,"data":"gaia-demo-plugin"}
{"authority":"arch-ad-management-test-1.隐藏.net","bytes_received":"0","bytes_sent":"87","downstream_local_address":"172.20.0.7:80","downstream_remote_address":"172.16.110.32:62646","duration":"23","istio_policy_status":"-","method":"GET","path":"/dubbo/hello/sayHello?name=xx","protocol":"HTTP/1.1","request_id":"fd9e0eb6-6592-48d1-8813-47090f53ca6e","requested_server_name":"-","response_code":"502","response_flags":"UPE","route_name":"gaia-demo-http2dubbo","start_time":"2024-06-25T09:35:50.099Z","trace_id":"9b7e2ea2ac0f1e6f670082e0f39a6fa9","upstream_cluster":"outbound|20880||providers:com.隐藏.demo.dubbo.api.DemoService::betaA.DEFAULT-GROUP.3e3dcada-7c4a-4728-b4c4-1cfdb5b7d2d1.nacos","upstream_host":"10.244.36.242:20880","upstream_local_address":"172.20.0.7:35284","upstream_service_time":"-","upstream_transport_failure_reason":"-","user_agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/125.0.0.0 Safari/537.36","x_forwarded_for":"172.16.110.32"}
`

**Comment by johnlanni** [comment]
com.隐藏.demo.dubbo.api.DemoService

这是啥，怎么还有中文

**Reply by paxxie2** [reply]
对找示例，一行行对比修改了。现在是 Cannot parse RpcResult type from buffer 了。 我重新提个issue
